src/lib/components/workflow/dropdown-filter/workflow-datetime-filter.svelte

```diff
@@ -0,0 +1,233 @@
+
+  let custom = false;
+  let value = 'All Time';
+  let timeField = 'StartTime';
+
+  onMount(() => {
+    const timeFilter = $workflowFilters.find(
+      (f) => f.attribute === 'StartTime' || f.attribute === 'CloseTime',
+    );
+    if (timeFilter) {
+      value = timeFilter.value;
+      timeField = timeFilter.attribute as string;
+    }
+  });
+
+  const getOtherFilters = () =>
+    $workflowFilters.filter(
+      (f) => f.attribute !== 'StartTime' && f.attribute !== 'CloseTime',
+    );
+
+  const onChange = (_value: string) => {
+    value = _value;
+    if (value === 'All Time') {
+      $workflowFilters = [...getOtherFilters()];
+      custom = false;
+    } else if (value === 'Custom') {
+      custom = true;
+    } else {
+      const filter = {
+        attribute: timeField,
+        value,
+        conditional: '=',
+        operator: '',
+        parenthesis: '',
+      };
+      $workflowFilters = [...getOtherFilters(), filter];
+      custom = false;
+    }
+
+    updateQueryParamsFromFilter($page.url, $workflowFilters, $workflowSorts);
+  };
+
+  const onTimeFieldChange = (field: 'StartTime' | 'CloseTime') => {
+    if (field !== timeField) {
+      timeField = field;
+      onChange(value);
+    }
+  };
+
+  let startDate = startOfDay(new Date());
+  let endDate = endOfDay(new Date());
+  let startTime = { hour: '', minute: '', second: '' };
+  let endTime = { hour: '', minute: '', second: '' };
+
+  const onStartDateChange = (d) => {
+    startDate = startOfDay(d.detail);
+  };
+
+  const onStartDateTimeChange = (d) => {
+    startTime = d.detail;
+  };
+
+  const onEndDateChange = (d) => {
+    endDate = startOfDay(d.detail);
+  };
+
+  const onEndDateTimeChange = (d) => {
+    endTime = d.detail;
+  };
+
+  const applyTimeChanges = (date: Date, time) => {
+    let _date = new Date(date);
+    if (time.hour) _date = addHours(_date, time.hour);
+    if (time.minute) _date = addMinutes(_date, time.minute);
+    if (time.second) _date = addSeconds(_date, time.second);
+    return _date;
+  };
+
+  const onApply = () => {
+    startDate = applyTimeChanges(startDate, startTime);
+    endDate = applyTimeChanges(endDate, endTime);
+    const filter = {
+      attribute: timeField,
+      value: `BETWEEN "${formatISO(startDate)}" AND "${formatISO(endDate)}"`,
+      conditional: '=',
+      operator: '',
+      parenthesis: '',
+      customDate: true,
+    };
+    $workflowFilters = [...getOtherFilters(), filter];
+
+    updateQueryParamsFromFilter($page.url, $workflowFilters, $workflowSorts);
+  };
+</script>
+
+<div class="flex items-center">
+  <Select
+    class="w-44 rounded"
+    id="time-range-filter"
+    placeholder="All Time"
+    unroundRight
+    keepOpen={true}
+    {value}
+    {onChange}
+  >
+    <div class="flex h-[320px] w-[400px] flex-col gap-8 bg-white p-4">
+      {#if custom}
+        <div class="flex flex-col">
+          <p class="text-sm">To</p>
+          <div class="flex gap-2">
+            <DatePicker
```

Comment:
The `DatePicker` gets cut off in the dropdown here. Not a super big deal, but maybe we want to make the dropdown bigger or have the `DatePicker` appear on top of the dropdown?

|![Screen Shot 2022-10-12 at 10 31 45 AM](https://user-images.githubusercontent.com/15069288/195399092-a83bb2e4-6501-4eb6-8b5a-2646ef3ca39e.png)|![Screen Shot 2022-10-12 at 10 33 06 AM](https://user-images.githubusercontent.com/15069288/195399144-ba18e0f0-d9d7-410d-ac3a-95f251deeaa0.png)|
|---|---|